java/statestore/src/main/java/sleeper/statestore/s3/S3FileInfoStore.java

```diff
@@ -352,24 +325,31 @@ public AllFileReferences getAllFileReferences() throws StateStoreException {
         }
     }
 
-    private static FileReferences references(String filename, List<FileInfo> references) {
+    private static FileReferences references(String filename, List<S3FileInfo> references) {
         Instant lastUpdateTime = references.stream()
-                .map(fileInfo -> Instant.ofEpochMilli(fileInfo.getLastStateStoreUpdateTime()))
+                .map(fileInfo -> Instant.ofEpochMilli(fileInfo.getLastUpdateTime()))
                 .max(Comparator.comparing(Function.identity())).orElseThrow();
         return new FileReferences(filename, lastUpdateTime, references.stream()
-                .filter(f -> f.getFileStatus().equals(FileInfo.FileStatus.ACTIVE))
+                .filter(f -> f.getFileStatus() == S3FileInfo.FileStatus.ACTIVE)
+                .map(S3FileInfo::getFileInfo)
                 .collect(Collectors.toUnmodifiableList()));
     }
 
-    private void updateFiles(Function<List<FileInfo>, List<FileInfo>> update, Function<List<FileInfo>, String> condition)
+    private void updateS3Files(Function<List<S3FileInfo>, List<S3FileInfo>> update, Function<List<S3FileInfo>, String> condition)
+            throws IOException, StateStoreException {
+        updateFiles(update, condition, this::readS3FileInfosFromParquet, this::writeS3FileInfosToParquet);
+    }
+
+    private <T> void updateFiles(Function<List<T>, List<T>> update, Function<List<T>, String> condition,
+                                 ReadFromParquet<T> readFromParquet, WriteToParquet<T> writeFilesToParquet)
```

Comment:
It looks like the type parameter here can be removed now, since it's always S3FileInfo?